src/Elastic.Apm.AspNetCore/ApmMiddleware.cs

```diff
@@ -88,6 +86,22 @@ public async Task InvokeAsync(HttpContext context)
 					Headers = responseHeaders
 				};
 
+				if (context.User?.Identity != null && context.User.Identity.IsAuthenticated && context.User.Identity != null
+					&& transaction.Context.User == null)
+				{
+					transaction.Context.User = new User { UserName = context.User.Identity.Name };
+
+					var idQuery = context.User.Claims.Where(n => n.Type == "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/nameidentifier");
+					var idClaims = idQuery.ToList();
+					if (idClaims.Any())
+						transaction.Context.User.Id = idClaims.FirstOrDefault()?.Value;
+
+					var mailQuery = context.User.Claims.Where(n => n.Type == "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress");

```

Comment:
Same here, in case of OAuth/OpenID - `email`

Reply:
Same as above.